### ISSUE TITLE
Consider iterating the dictionary directly instead of calling .keys() don't work with dict get method

### ISSUE THREAD

**Issue by Monk3yDev** [PyCQA/pylint#3085]
### Steps to reproduce
```python
"""Module"""

# Simple check. All is fine
WORKING = {"1": "Is ok"}
for key in WORKING.keys():
    print(key)

# Get key 2. Return value WORKING dict
# Lint message appear
ANOTHER_EXAMPLE = {"2": WORKING}["2"]
for key in ANOTHER_EXAMPLE.keys():
    print(key)

# Try get key 3. Get return default value which is WORKING dict
# No lint is raised
NOT_WORKING = {"2": WORKING}.get("3", WORKING)
for key in NOT_WORKING.keys():
    print(key)
```

### Current behavior
![lint_msg](https://user-images.githubusercontent.com/8431461/64069531-645de780-cc4b-11e9-8cd5-555b8f35f38c.png)
![lint_msg_info](https://user-images.githubusercontent.com/8431461/64069532-66c04180-cc4b-11e9-9534-9e2fd490ea68.png)
### Expected behavior
All 3 for loops should raise lint message.

### pylint --version output
__main__.py 2.3.1
astroid 2.2.5
Python 3.7.4 (tags/v3.7.4:e09359112e, Jul  8 2019, 19:29:22) [MSC v.1916 32 bit (Intel)]

**Comment by PCManticore** [comment]
Thank you, I can reproduce the issue. If we remove the last two iterations, the check is correctly emitted, but it is not emitted in conjunction with the other two.

**Comment by PCManticore** [comment]
On a second look we do not emit the check for the last occurrence as this check relies on inference to figure out that the owner of `keys()` is an actual dictionary. In this particular case we cannot infer that the owner of `keys()` is a dictionary, since `NOT_WORKING` is constructed via a `.get()` call. If our inference engine would improve and would properly infer `NOT_WORKING` this check will be emitted as well.
